Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -153,3 +160,222 @@ theorem LSeries_eq_mul_integral_of_nonneg (f : ℕ → ℝ) {r : ℝ} (hr : 0 
+    replace ht : (t : ℂ) ≠ 0 := ne_zero_of_one_lt_re ht
+    rw [sub_mul, cpow_sub _ _ ht, cpow_one, mul_assoc, mul_div_cancel₀ _ ht]
+  · rw [norm_mul, show ((s : ℂ) - 1) * s = ((s - 1) * s : ℝ) by simp, norm_real,
+      Real.norm_of_nonneg hs']
+    exact mul_le_mul_of_nonneg_left (norm_integral_le_integral_norm _) hs'
+  · congr 1
+    refine setIntegral_congr_fun measurableSet_Ioi fun t ht ↦ ?_
+    replace ht : 0 ≤ t := zero_le_one.trans ht.le
+    rw [norm_mul, show (-(s : ℂ) - 1) = (-s - 1 : ℝ) by simp, ← ofReal_cpow ht, norm_real,
+      Real.norm_of_nonneg (Real.rpow_nonneg ht _)]
+  · rw [← Set.Ioc_union_Ioi_eq_Ioi hT₁, setIntegral_union (Set.Ioc_disjoint_Ioi le_rfl)
+      measurableSet_Ioi]
+    · exact h₁.mono_set <| Set.Ioc_subset_Ioi_self.trans Set.Ioi_subset_Ici_self
+    · exact h₁.mono_set <| Set.Ioi_subset_Ici_self.trans <| Set.Ici_subset_Ici.mpr hT₁
+  · rw [mul_add, ← mul_assoc, mul_comm s]
+    exact add_le_add_right (mul_le_mul_of_nonneg_left h₄ hs') _
+  · gcongr
+    exact LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₂ h₀ hε hs hT₁
+      fun t ht ↦ (hT t ht.le).le
+
+include hlim hfS in
+theorem LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div :
+    Tendsto (fun s : ℝ ↦ (s - 1) * LSeries f s) (𝓝[>] 1) (𝓝 l) := by
+  refine Metric.tendsto_nhdsWithin_nhds.mpr fun ε hε ↦ ?_
+  have hε6 : 0 < ε / 6 := by positivity
+  have hε3 : 0 < ε / 3 := by positivity
+  obtain ⟨C, hC₁, hC₂⟩ := LSeries_tendsto_sub_mul_nhds_one_of_tendsto_sum_div_aux₃ hlim hfS hε6
+  have lim1 : Tendsto (fun s ↦ (s - 1) * s * C) (𝓝[>] 1) (𝓝 0) := by
+    have : ContinuousAt (fun s ↦ (s - 1) * s * C) 1 := by fun_prop
+    convert tendsto_nhdsWithin_of_tendsto_nhds this.tendsto
+    rw [sub_self, zero_mul, zero_mul]
+  have lim2 : Tendsto (fun s : ℝ ↦ s * l) (𝓝[>] 1) (𝓝 l) := by
+    have : ContinuousAt (fun s : ℝ ↦ s * l) 1 := by fun_prop
+    convert tendsto_nhdsWithin_of_tendsto_nhds this.tendsto
+    rw [Complex.ofReal_one, one_mul]
+  rw [Metric.tendsto_nhdsWithin_nhds] at lim1 lim2
+  obtain ⟨δ₁, _, hδ₁⟩ := lim1 _ hε3
+  obtain ⟨δ₂, _, hδ₂⟩ := lim2 _ hε3
+  refine ⟨min 1 (min δ₁ δ₂), by positivity, fun s hs₁ hs₂ ↦ ?_⟩
+  specialize hC₂ s hs₁
+  specialize hδ₁ hs₁ <| hs₂.trans_le <| (min_le_right _ _).trans (min_le_left _ _)
+  specialize hδ₂ hs₁ <| hs₂.trans_le <| (min_le_right _ _).trans (min_le_right _ _)
+  rw [dist_eq_norm] at hδ₁ hδ₂ hs₂ ⊢
+  rw [sub_zero, Real.norm_of_nonneg (mul_nonneg
+    (mul_nonneg (sub_nonneg.mpr hs₁.le) (zero_lt_one.trans hs₁).le) hC₁)] at hδ₁
+  have sle2 : s < 2 := by
+    have := (abs_lt.mp <| Real.norm_eq_abs _ ▸ (hs₂.trans_le (min_le_left _ _))).2
+    rwa [sub_lt_iff_lt_add', one_add_one_eq_two] at this
+  calc
+    _ ≤ ‖(s - 1) * LSeries f s - l * s‖ + ‖l * s - l‖ := norm_sub_le_norm_sub_add_norm_sub _ _ _
+    _ < (s - 1) * s * C + s * (ε / 6) + (ε / 3) := add_lt_add_of_le_of_lt hC₂ (by rwa [mul_comm])
+    _ ≤ (ε / 3) + s * (ε / 6) + (ε / 3) := by gcongr
+    _ < (ε / 3) + (ε / 3) + (ε / 3) := ?_
```

Comment:
Is it possible that these fiddly inequality manipulations could be avoided by using a "softer" approach working with filters, rather than rewriting everything in terms of epsilons?

Reply:
Yeah, there is a much simpler proof using `limsup` but I found it painful to work with it in the past. Still, I can give it another try.

Reply:
I was thinking more of whether something like `Eventually.and` or `filter_upwards` etc could help you, but that was a rather naive comment before I had a chance to really read the proof, and probably it doesn't work. Sorry for the noise!

Reply:
Anyway, I decided I would not let the `limsup` API bully me so I wrote a new proof using `limsup`. Sorry about the extra work :)